e2e/testcases/helm_sync_test.go

```diff
@@ -205,47 +177,58 @@ service:
+
+	cmName := "helm-watch-config-map"
+	nt.T.Log("Apply valid ConfigMap that is not immutable (which should not be allowed)")
+	f := false
+	cm0 := fake.ConfigMapObject(core.Name(cmName), core.Namespace(configsync.ControllerNamespace))
+	cm0.Immutable = &f
+	cm0.Data = map[string]string{"something-else.yaml": `
+image:
+  digest: sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e
+  pullPolicy: Never
+`,
+	}
+	nt.T.Cleanup(func() {
+		if err := nt.KubeClient.Delete(cm0); err != nil {
+			nt.T.Log(err)
+		}
+	})
+	if err := nt.KubeClient.Create(cm0); err != nil {
 		nt.T.Fatal(err)
 	}
-	var expectedCPURequest string
-	var expectedCPULimit string
-	var expectedMemoryRequest string
-	var expectedMemoryLimit string
-	if nt.IsGKEAutopilot {
-		expectedCPURequest = "250m"
-		expectedCPULimit = expectedCPURequest
-		expectedMemoryRequest = "512Mi"
-		expectedMemoryLimit = expectedMemoryRequest
-	} else {
-		expectedCPURequest = "150m"
-		expectedCPULimit = "1"
-		expectedMemoryRequest = "250Mi"
-		expectedMemoryLimit = "300Mi"
+	nt.WaitForRootSyncStalledError(rs.Namespace, rs.Name, "Validation", "KNV1061: RootSyncs must only reference ConfigMaps that are immutable; ConfigMap \"helm-watch-config-map\" in namespace \"config-management-system\" is not immutable")
+
+	tr := true
+	nt.T.Log("Apply the ConfigMap with values to the cluster with incorrect data key")
+	cm1 := fake.ConfigMapObject(core.Name(cmName), core.Namespace(configsync.ControllerNamespace))
+	cm1.Immutable = &tr
+	cm1.Data = map[string]string{"something-else.yaml": `
+image:
+  digest: sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e
+  pullPolicy: Never
+`,
 	}
-	if err := nt.Validate("my-wordpress", "wordpress", &appsv1.Deployment{}, containerImagePullPolicy("Always"),
-		testpredicates.HasCorrectResourceRequestsLimits("wordpress",
-			resource.MustParse(expectedCPURequest),
-			resource.MustParse(expectedCPULimit),
-			resource.MustParse(expectedMemoryRequest),
-			resource.MustParse(expectedMemoryLimit)),
-		testpredicates.HasExactlyImage("wordpress", "bitnami/wordpress", "", "sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e"),
-		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_USERNAME", "test-user-1"),
-		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_EMAIL", "test-user@example.com"),
-		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_1", "val1"),
-		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_2", "val2")); err != nil {
+	if err := nt.KubeClient.Update(cm1); err != nil {
 		nt.T.Fatal(err)
 	}
+	nt.WaitForRootSyncStalledError(rs.Namespace, rs.Name, "Validation", "KNV1061: RootSync field spec.helm.valuesFileRefs.valuesFile must specify an existing data key in the referenced ConfigMap; data key \"values.yaml\" not found in ConfigMap \"helm-watch-config-map\"")
```

Comment:
`RootSyncs must reference valid ConfigMaps in spec.helm.valuesFileRefs: ConfigMap \"helm-watch-config-map\" in namespace \"config-management-system\" does not contain the data key \"values.yaml\".

Reply:
updated the error messages